daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto

```diff
@@ -951,6 +951,31 @@ message Expr {
     Expr interface_expr = 3;
   }
 
+  // Obtain the type representation of a contract through an interface
+  // *Available in versions >= 1.dev*
+  message ToTypeRep {
+    // interface type
+    TypeConName interface_type = 1;
```

Comment:
does this encoding make sense? AFAICT we want `TO_TYPE_REP` to return the _template_ type for a contract, so we don't really care about the interface. Maybe this could even be a builtin?

Reply:
Interested to hear what @remyhaemmerle-da thinks.
I think it does make sense for a few reasons:
1. This isn’t going to work for arbitrary types. I believe the current implementation only works for record types. I’d rather have it be a type error than a runtime error if you misuse it.
2. If we expect interface_type to be an interface, we should also check for that in the typechecker as mentioned in my comment above. We cannot do that if it’s a type variable.

Reply:
- We care of the interface for typing, I think we do not want to run this over arbitrary record/template, but only on those template that extend interface. Otherwise this will make our life more complicate if we want to erase type inside Speedy. 

- I prefer to keep the builtin for think that does not depend of the LF program. 

- We probably need a more explicit name for this primitive.

Reply:
thank you both! I agree with all your points.

> we should also check for that in the typechecker as mentioned in my comment above

coming up!

> We probably need a more explicit name for this primitive

that's a good point, do you have any ideas in mind?

Reply:
I am very bad at giving name. 
maybe `EInterfaceToTypeRep`

Reply:
@cocreature suggested `EInterfaceTemplateTypeRep` on a different thread, I'll go ahead with that one :)